storagenode/storagenodedb/database.go

```diff
@@ -232,22 +205,35 @@ func (db *DB) Close() error {
 		db.ndb.Close(),
 		db.adb.Close(),
 
-		db.versionsDB.Close(),
-		db.v0PieceInfoDB.Close(),
-		db.bandwidthDB.Close(),
-		db.ordersDB.Close(),
-		db.pieceExpirationDB.Close(),
-		db.pieceSpaceUsedDB.Close(),
-		db.reputationDB.Close(),
-		db.storageUsageDB.Close(),
-		db.usedSerialsDB.Close(),
-		db.satellitesDB.Close(),
+		db.closeDatabases(),
 	)
 }
 
-// Versions returns the instance of the versions database.
-func (db *DB) Versions() SQLDB {
-	return db.versionsDB
+// closeDatabases closes all the SQLite database connections and removes them from the associated maps.
+func (db *DB) closeDatabases() error {
+	var err error
+
+	for k := range db.sqlDatabases {
+		_ = errs.Combine(err, db.closeDatabase(k))
+	}
+	return err
+}
+
+// closeDatabase closes the specified SQLite database connections and removes them from the associated maps.
+func (db *DB) closeDatabase(dbName string) (err error) {
+	if conn, ok := db.sqlDatabases[dbName]; ok {
```

Comment:
Why would a database be double closed? This seems like an error prone way to handle life-cycles.

In general for lifecycles: 
1. For every open, there should be exactly one close call.
2. There should be no early closing of shared resources... e.g. which means if there's a resource `X` that is shared by different things only the last `close` should actually close the resource `X`... see `sync2` package with multitee things to ensure this is the case.

Reply:
I'm aware of lifecycles and objects scope. The original design of the `database` object and its child objects had poor encapsulation to begin with that caused all kinds of bugs and friction. I tried to improve this incrementally in this PR. I don't think I introduced an early close or double close have I? If I have that wasn't intended 😄

Reply:
If there's no double close you can replace this code with:

```
conn, ok := db.sqlDatabases[dbName]
if !ok { return Error.New("double close") }
```

Reply:
Yeah looks, like it's correct and can be replaced with:
```
// closeDatabase closes the specified SQLite database connections and removes them from the associated maps.
func (db *DB) closeDatabase(dbName string) (err error) {
	conn, ok := db.sqlDatabases[dbName]
	if !ok {
		return ErrDatabase.New("double close")
	}
	delete(db.sqlDatabases, dbName)
	return ErrDatabase.Wrap(conn.Close())
}
```